server/models/User.js

```diff
@@ -27,8 +27,25 @@ module.exports = function(bookshelf){
 
     onCreate: function(model, attrs, options) {
       var self = this;
+
+      // self.get('password') returns undefined if password doesnt exist
```

Comment:
So a user was being created and i guess bcrypt was just encrypting 'undefined', because a user would always be created with a password before I changed this.